controllers/release/release_adapter.go

```diff
@@ -155,19 +159,37 @@ func (a *Adapter) EnsureReleasePipelineStatusIsTracked() (results.OperationResul
 	return results.ContinueProcessing()
 }
 
+// EnsureTargetContextIsSet is an operation that will ensure that the targetContext of the adapter is not nil. It will
+// set it to a context from the workspace in the ReleaseLink if one is defined or to its own context if not.
+func (a *Adapter) EnsureTargetContextIsSet() (results.OperationResult, error) {
+	releaseLink, err := a.getReleaseLink()
+	if err != nil {
+		return results.RequeueWithError(err)
+	}
+
+	if a.targetContext == nil {
+		if releaseLink.Spec.Target.Workspace == "" {
+			a.targetContext = a.context
+		} else {
+			a.targetContext = logicalcluster.WithCluster(a.context, logicalcluster.New(releaseLink.Spec.Target.Workspace))
```

Comment:
reminder to myself to test what happens if I run a get operation using a context for a cluster that doesn't exist

Reply:
That's a good test. I'm expecting it to simply fail and those errors are already catched. In any case, I agree on this 
 being a good thing to test.